test-case/check-volume-levels.sh

```diff
@@ -0,0 +1,341 @@
+
+    # Start sine playback
+    aplay_helper "$APLAY_WAV" & aplayPID=$!
+    pause
+
+    # Do capture tests
+    run_test_1
+    run_test_2
+    run_test_3
+
+    # Stop sine playback and do cleanup
+    pause
+    dlogi "The test procedure is now complete. Killing the aplay process $aplayPID."
+    kill $aplayPID
+
+    # Measure, delete unnecessary wav files if passed
+    if measure_levels; then
+	dlogi "Deleting temporary files"
+	rm -f "$APLAY_WAV" "$ARECORD_WAV1" "$ARECORD_WAV2" "$ARECORD_WAV3"
+    fi
+}
+
+#
+# Test #1 volume and mute switch controls
+#
+
+#  0...1s channels 1-4 max gain
+#  1...2s channels 1-4 different gains
+#  2...3s channels 1-4 nominal gain
+#  3...4s channels 1-4 muted
+#  4...5s channels 1-4 nominal gain
+#  5...7s channels 1-4 muted
+#  7...8s channels 1-4 max gain
+#  8...9s channels 1-4 muted gain
+#  9..10s channels 1-4 min gain
+# 10..11s channels 1-4 different gains
+run_test_1 () {
+    cset_max; cset_unmute; pause
+
+    arecord_helper "$ARECORD_WAV1" "$CAP_CHANNELS" 11 & arecordPID=$!
+    pause
+    cset_volume_diff1 "$CAP_CHANNELS"; pause
+    cset_nom; pause
+    cset_mute; pause
+    cset_unmute; pause
+    cset_mute; pause
+    cset_max; pause
+    cset_unmute; pause
+    cset_mutevol; pause
+    cset_min; pause
+    cset_volume_diff2 "$CAP_CHANNELS"
+
+    # Wait for arecord process to complete
+    dlogi "Waiting $arecordPID"
+    wait $arecordPID
+    dlogi "Ready."
+}
+
+#
+# Test #2, check gains preservation from previous to next
+#
+
+# 0..1s channels 1-4 previous gains
+# 1..2s channels 1-4 different mute switches
+# 2..3s channels 1-4 all muted
+run_test_2 () {
+    arecord_helper "$ARECORD_WAV2" "$CAP_CHANNELS" 4 & arecordPID=$!
+    pause
+    cset_nom; cset_mute_diff1 "$CAP_CHANNELS"; pause
+    cset_mute; pause
+
+    dlogi "Waiting $arecordPID"
+    wait $arecordPID
+    dlogi "Ready."
+}
+
+#
+# Test #3, test mute switch preservation from previous to next
+#
+
+# 0..1s channels 1-4 muted
+# 1..2s channels 1-4  nominal gain
+# 2..3s different mute switches
+# 3..4s channels 1-4 nominal gain
+run_test_3 () {
+    arecord_helper "$ARECORD_WAV3" "$CAP_CHANNELS" 4 & arecordPID=$!
+    pause
+    cset_nom; pause
+    cset_mute_diff2 "$CAP_CHANNELS"; pause
+    cset_unmute
+
+    dlogi "Waiting $arecordPID"
+    wait $arecordPID
+    dlogi "Ready."
+}
+
+#
+# Helper functions
+#
+
+cset_nom () {
+    dlogi "Set nominal volume"
+    amixer cset name="$CAP_VOLUME" $VOL_NOM
+}
+
+cset_max () {
+    dlogi "Set maximum volume"
+    amixer cset name="$CAP_VOLUME" $VOL_MAX
+}
+
+cset_min () {
+    dlogi "Set minimum volume"
+    amixer cset name="$CAP_VOLUME" $VOL_MIN
+}
+
+cset_mutevol () {
+    dlogi "Set mute via volume"
+    amixer cset name="$CAP_VOLUME" $VOL_MUTE
+}
+cset_mute () {
+    dlogi "Set mute via switch"
+    amixer cset name="$CAP_SWITCH" off
+}
+
+cset_unmute () {
+    dlogi "Set unmute"
+    amixer cset name="$CAP_SWITCH" on
+}
+
+pause () {
+    dlogi "Pausing"
```

Comment:
The wording "pause" is somewhat confusion, we are not aim to pause any playback/capture stream, maybe 'sleep_1s'?

Reply:
The test script really pauses execution so I prefer this. Just sleep is a reserved word and don't want to hard code with the name the length to 1s. It could be adjusted later to more than 1s for better accuracy or less than 1s for lower test time.

Reply:
`nap()` ? :-)

Reply:
yes, either 'nap' / 'nop' or 'idle' sound better to me, or even exec_pause(), as we have test cases naming xxx_pause_release_xxx, I thought we are pausing the playback/capture pipeline at my first glance to the PR.

Reply:
nap sounds good to me!